server/src/main/scala/FileRequester.scala

```diff
@@ -50,23 +70,47 @@ class FileRequester(mongoConfig: MongoConfig) {
 
   def create(x: FileDocument): Future[InsertOneResult] = {
     logger.info("Performing a create on {}", x.path)
+    x.version = Some(1)
     current_coll.insertOne(x).toFuture()
   }
 
-  def update(x: FileDocument): Future[Option[UpdateResult]] = {
+  //update is a new version of the file
+  def update(x: FileDocument): Future[InsertOneResult] = {
     logger.info("Performing an update on {}", x.path)
-    val updatedFile: Observable[UpdateResult] = current_coll
-      .find(equal("path", x.path))
+    val latestVersion = findLatestVersion(x.path)
+    println(latestVersion)
+    x.version = Some(latestVersion.get + 1)
+    current_coll.insertOne(x).toFuture()
+  }
+
+  //delete is a new version of the file with deleted boolean set to True
+  def delete(x: FileDocument): Future[InsertOneResult] = {
+    logger.info("Performing a delete on {}", x.path)
+    val latestVersion = findLatestVersion(x.path)
+    x.version = Some(latestVersion.get + 1)
+    x.deleted = true
+    current_coll.insertOne(x).toFuture()
```

Comment:
Why you didn't update the current document instead of creating a new one? 

Because if you delete a document `x`, with the current behavior, you'll create a duplicate ?

Reply:
because delete is a **new** version of the file with deleted boolean set to True